net/src/Sails.Remoting/RemotingViaSubstrateClient.cs

```diff
@@ -1,30 +1,63 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using EnsureThat;
 using Sails.Remoting.Abstractions;
 using Sails.Remoting.Options;
 using Substrate.Gear.Api.Generated;
+using Substrate.Gear.Api.Generated.Model.frame_system;
 using Substrate.Gear.Api.Generated.Model.gprimitives;
+using Substrate.Gear.Api.Generated.Model.vara_runtime;
+using Substrate.Gear.Api.Generated.Storage;
+using Substrate.Gear.Client;
+using Substrate.Gear.Client.Model.Types;
+using Substrate.Gear.Client.Model.Types.Base;
 using Substrate.NetApi.Model.Extrinsics;
+using Substrate.NetApi.Model.Types;
+using Substrate.NetApi.Model.Types.Base;
+using Substrate.NetApi.Model.Types.Primitive;
+using EnumGearEvent = Substrate.Gear.Api.Generated.Model.pallet_gear.pallet.EnumEvent;
 using GasUnit = Substrate.NetApi.Model.Types.Primitive.U64;
+using GearEvent = Substrate.Gear.Api.Generated.Model.pallet_gear.pallet.Event;
+using MessageQueuedGearEventData = Substrate.NetApi.Model.Types.Base.BaseTuple<
+    Substrate.Gear.Api.Generated.Model.gprimitives.MessageId,
+    Substrate.Gear.Api.Generated.Model.sp_core.crypto.AccountId32,
+    Substrate.Gear.Api.Generated.Model.gprimitives.ActorId,
+    Substrate.Gear.Api.Generated.Model.gear_common.@event.EnumMessageEntry>;
 using ValueUnit = Substrate.NetApi.Model.Types.Primitive.U128;
 
 namespace Sails.Remoting;
 
 internal sealed class RemotingViaSubstrateClient : IDisposable, IRemoting
 {
-    public RemotingViaSubstrateClient(RemotingViaSubstrateClientOptions options)
+    /// <summary>
+    /// Creates an instance implementing the <see cref="IRemoting"/> interface via <see cref="SubstrateClientExt"/>
+    /// with initial account for signing transactions.
+    /// </summary>
+    /// <param name="options"></param>
+    /// <param name="signingAccount"></param>
+    public RemotingViaSubstrateClient(
+        RemotingViaSubstrateClientOptions options,
+        Account signingAccount)
```

Comment:
My thoughts
Remoting should be a lightweight wrapper over SubstrateClient and extend it with Account data.
The SubstrateClient itself should be Singleton (if possible),  and its lifecycle should be managed from RemotingProvider.

Then, Remoting will be quickly created from the provider and use a shared SubstrateClient, like `Clone`